src/tests/protocol.rs

```diff
@@ -0,0 +1,54 @@
+#[cfg(all(feature = "shuttle", test))]
+mod randomized {
```

Comment:
This doesn't need a module here, does it?  You could make this:

```suggestion
#![cfg(all(feature = "shuttle", test))]
```